main.go

```diff
@@ -20,478 +20,15 @@
-
-		// Run.
-		err = daemonize.Run(path, args, env, os.Stdout)
-		if err != nil {
-			return fmt.Errorf("daemonize.Run: %w", err)
-		}
-		logger.Infof(SuccessfulMountMessage)
-		return err
-	}
-
-	// The returned error is ignored as we do not enforce monitoring exporters
-	monitor.EnableStackdriverExporter(flags.StackdriverExportInterval)
-	monitor.EnableOpenTelemetryCollectorExporter(flags.OtelCollectorAddress)
-
-	// Mount, writing information about our progress to the writer that package
-	// daemonize gives us and telling it about the outcome.
-	var mfs *fuse.MountedFileSystem
-	{
-		mfs, err = mountWithArgs(bucketName, mountPoint, flags, mountConfig)
-
-		// This utility is to absorb the error
-		// returned by daemonize.SignalOutcome calls by simply
-		// logging them as error logs.
-		callDaemonizeSignalOutcome := func(err error) {
-			if err2 := daemonize.SignalOutcome(err); err2 != nil {
-				logger.Errorf("Failed to signal error to parent-process from daemon: %v", err2)
-			}
-		}
-
-		markSuccessfulMount := func() {
-			// Print the success message in the log-file/stdout depending on what the logger is set to.
-			logger.Info(SuccessfulMountMessage)
-			callDaemonizeSignalOutcome(nil)
-		}
-
-		markMountFailure := func(err error) {
-			// Printing via mountStatus will have duplicate logs on the console while
-			// mounting gcsfuse in foreground mode. But this is important to avoid
-			// losing error logs when run in the background mode.
-			logger.Errorf("%s: %v\n", UnsuccessfulMountMessagePrefix, err)
-			err = fmt.Errorf("%s: mountWithArgs: %w", UnsuccessfulMountMessagePrefix, err)
-			callDaemonizeSignalOutcome(err)
-		}
-
-		if err != nil {
-			markMountFailure(err)
-			return err
-		}
-		if !isDynamicMount(bucketName) {
-			switch flags.ExperimentalMetadataPrefetchOnMount {
-			case config.ExperimentalMetadataPrefetchOnMountSynchronous:
-				if err = callListRecursive(mountPoint); err != nil {
-					markMountFailure(err)
-					return err
-				}
-			case config.ExperimentalMetadataPrefetchOnMountAsynchronous:
-				go func() {
-					if err := callListRecursive(mountPoint); err != nil {
-						logger.Errorf("Metadata-prefetch failed: %v", err)
-					}
-				}()
-			}
-		}
-		markSuccessfulMount()
-	}
-
-	// Let the user unmount with Ctrl-C (SIGINT).
-	registerSIGINTHandler(mfs.Dir())
-
-	// Wait for the file system to be unmounted.
-	err = mfs.Join(context.Background())
-
-	monitor.CloseStackdriverExporter()
-	monitor.CloseOpenTelemetryCollectorExporter()
-
-	if err != nil {
-		err = fmt.Errorf("MountedFileSystem.Join: %w", err)
-		return
-	}
-
-	return
-}
-
-func run() (err error) {
-	// Set up the app.
-	app := newApp()
-
-	var appErr error
-	app.Action = func(c *cli.Context) {
-		appErr = runCLIApp(c)
-	}
-
-	// Run it.
-	err = app.Run(os.Args)
-	if err != nil {
-		return
-	}
-
-	err = appErr
-	return
-}
-
-// Handle panic if the crash occurs during mounting.
-func handlePanicWhileMounting() {
+func logPanic() {
```

Comment:
nit: logPanicWhileMounting

Reply:
This is a catch-all and not limited to mounting. We would like to log even for non-mounting cases such as showing the version or generating the config file from the given set of flags. That's why I think we should rename it.